readthedocs/doc_builder/python_environments.py

```diff
@@ -63,13 +68,27 @@ def delete_existing_venv_dir(self):
             ))
             shutil.rmtree(venv_dir)
 
-    def install_package(self):
-        if (self.config.python.install_with_pip or
-                getattr(settings, 'USE_PIP_INSTALL', False)):
+    def install_requirements(self):
+        for install in self.config.python.install:
+            self._run_install_step(install)
+
+    def _run_install_step(self, install):
+        if isinstance(install, PythonInstallRequirements):
+            return self.install_requirements_file(install)
+        if isinstance(install, PythonInstall):
+            return self.install_package(install)
+
+    def install_package(self, install):
```

Comment:
I'd love to see good docstring for all these new methods ;)

Reply:
I've added some docstrings, not sure if are the best, I wasn't sure what else to write there

Reply:
I try to think on what the reader should know about this method without reading the "whole" source code or, by coming to this method from another part of the code (where it's used). With that in mind, I usually try to write a description.

Besides, I also try to document the arguments with:

```
:param install: some description about it
:type install: readthedocs.config.models.PythonInstall (or whatever)
```

With that information, it's easier to follow and understand a method IMHO.

Reply:
I wrote `is_obsolete` in that same file, for example.